### PR TITLE
Enable kola on arm64 machines

### PR DESCRIPTION
**Pull request by glevand**
Allows kola to run on arm64 hosts.  Also includes a patch that increases the debug output of dnsmasq.

### CODE REVIEW

File: platform/local/dnsmasq.go [hunk 1]
```diff
@@ -79,13 +91,11 @@ dhcp-host={{.HardwareAddr}}{{template "ips" .DHCPv4}}{{template "ips" .DHCPv6}}
...
 {{define "ips"}}{{range .}}{{printf ",%s" .IP}}{{end}}{{end}}
-`))
-
-const (
-	numInterfaces = 16
-	numSegments   = 3
+`
 )
 
+var plog = capnslog.NewPackageLogger("github.com/coreos/mantle", "local")
```
Comment (bgilbert) [inline platform/local/dnsmasq.go]: The second argument should be `platform/local`.

File: kola/harness.go [hunk 2]
```diff
@@ -369,7 +369,7 @@ func runTest(h *harness.H, t *register.Test, pltfrm string) {
 
 // architecture returns the machine architecture of the given platform.
 func architecture(pltfrm string) string {
-	nativeArch := "amd64"
+	nativeArch := system.PortageArch()
```
Comment (bgilbert) [inline kola/harness.go]: If kola is running on arm64 but `pltfrm` is e.g. `aws`, `architecture` should still return `amd64`.  It looks like the right place to fix the default selection is in `sdk.DefaultBoard()`.  That will fix `QEMUOptions.Board` via `defaultTargetBoard` in `cmd/kola/options.go`.

File: platform/machine/qemu/cluster.go [hunk 3]
```diff
@@ -134,12 +134,14 @@ func (qc *Cluster) NewMachine(userdata *conf.UserData) (platform.Machine, error)
...
 			"-machine", "accel=kvm",
 			"-cpu", "host",
+			"-m", "1024",
 		}
 	case "arm64-usr":
 		qmCmd = []string{
 			"qemu-system-aarch64",
 			"-machine", "virt",
 			"-cpu", "cortex-a57",
+			"-m", "2048",
```
Comment (bgilbert) [inline platform/machine/qemu/cluster.go]: Just to double-check, this is necessary with images built with `./build_image prod`?  On amd64, dev images seem to need more than 1024 MB (https://github.com/coreos/bugs/issues/1594) but prod images don't.

File: kola/harness.go (hunk 2 above)
Comment (glevand) [inline kola/harness.go]: OK, please check what I have in the updated patch.

File: platform/machine/qemu/cluster.go (hunk 3 above)
Comment (glevand) [inline platform/machine/qemu/cluster.go]: Yes, the prod image will run out of memory.  Strange behavior and unexpected errors start to occur.

**Comment by glevand** [comment]
Rebase to latest.

**Comment by glevand** [comment]
In the current patch set the only change I've made is for sdk.DefaultBoard() to use system.PortageArch() for its default value, which will be the host machine's arch (from runtime.GOARCH).  QEMUOptions.Board should now be set to either a user specified `--board` value, or a default of the host machine.

kola.architecture() should now return amd64 for gce, aws, and esx, and either the `--board` value or the host machine's arch for qemu.

With my new patch `kola/harness: Use PacketOptions.Board` packet will have the same value as qemu, either the `--board` value or the host machine's arch.  Does it make more sense to have a default of amd64 for packet?  It doesn't seem it would be easy to arrange that though.

**Comment by bgilbert** [comment]
> Does it make more sense to have a default of amd64 for packet? It doesn't seem it would be easy to arrange that though.

I think it's fine to default Packet to the host platform.  We should arguably switch the default `--packet-plan` to `baremetal_2a` on ARM, but we can leave that for another day.

**Pull request merged**